java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/breakpoints/LineBreakpointImpl.java

```diff
@@ -337,32 +343,35 @@ protected void classLoaded (List<ReferenceType> referenceTypes) {
                     continue;
                 }
                 // Submit the breakpoint for the lowest location on the line only:
-                Location location = locations.get(0);
-                com.sun.jdi.Method m0 = location.method();
-                for (int li = 1; li < locations.size(); li++) {
-                    Location l = locations.get(li);
-                    if (l.codeIndex() < location.codeIndex()) {
-                        if (l.method().equals(m0)) {
-                            // Assure that we're still in the same method
-                            location = l;
-                        }
-                    }
-                }
+                /* If location contains multiple event breakpoint for the same line 
+                (e.g lambda expression along with any function calling 
+                map(n -> foo(
+                        boo(n)))) 
+                then take the lowest location from each type of event*/
+                Collection<Location> lowestLocationList = locations
+                        .stream()
+                        .collect(groupingBy(loc->loc.method().toString(),
+                                LinkedHashMap::new,
+                                collectingAndThen(minBy(Comparator.comparingDouble(Location::codeIndex)), 
```

Comment:
It's better to use `Comparator.comparingLong`

Reply:
comments addressed.

Reply:
Please squash the changes before merging